helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java

```diff
@@ -144,6 +168,94 @@
             .setUser(username)
             .setEndpoints("http://" + deploymentAddress)
             .build());
+
+    connectToWatchdog(dockerHost.address(), watchdogPort);
+
+    try {
+      start();
+    } catch (TimeoutException e) {
+      Throwables.propagate(e);
+    }
+  }
+
+  static synchronized HeliosDeployment getOrCreateHeliosSoloDeployment() {
+    if (soloDeployment == null) {
+      // TODO (dxia) remove checkForNewImages(). Set here to prevent using
+      // spotify/helios-solo:latest from docker hub
+      soloDeployment = HeliosSoloDeployment.fromEnv().checkForNewImages(false).build();
+    }
+    return soloDeployment;
+  }
+
+  private static void connectToWatchdog(final String host, final int port) {
+    final ExecutorService executorService = Executors.newSingleThreadExecutor(
+        new ThreadFactoryBuilder()
+            .setNameFormat("helios-solo-watchdog-%d")
+            .setDaemon(true).build());
+
+    log.info("Connecting to helios-solo watchdog at {}:{}", host, port);
+
+    executorService.submit(new Runnable() {
```

Comment:
might make sense to break this anonymous class into it's own class to make it easier to test on it's own